src/SourceGen/src/Core/AppBuilderGenerator.cs

```diff
@@ -0,0 +1,406 @@
+		const string MauiUIApplicationDelegateFullyQualifiedName = "global::Microsoft.Maui.MauiUIApplicationDelegate";
+		const string UIKitUIApplicationDelegateFullyQualifiedName = "global::UIKit.UIApplicationDelegate";
+		const string AndroidAppActivityFullyQualifiedName = "global::Android.App.ActivityAttribute";
+		const string MauiAndroidApplicationFullyQualifiedName = "global::Microsoft.Maui.MauiApplication";
+		const string AndroidAppApplicationFullyQualifiedName = "global::Android.App.Application";
+		const string AndroidMainLauncherAttributeName = "MainLauncher";
+		const string AppDelegateDefaultClassName = "MauiAppDelegate";
+		const string MainActivityDefaultClassName = "MauiMainActivity";
+		const string ProgramDefaultClassName = "MauiProgram";
+		const string MainMethodName = "Main";
+		const string AndroidApplicationDefaultClassName = "MauiApplication";
+
+		public void Initialize(GeneratorInitializationContext context)
+		{
+			context.RegisterForSyntaxNotifications(() => new AppStartupSyntaxReceiver());
+		}
+
+		static bool HasCreateMauiAppOverride(GeneratorSyntaxContext context, ClassDeclarationSyntax classDeclarationSyntax)
+		{
+			foreach (var m in classDeclarationSyntax.Members)
+			{
+				if (m is MethodDeclarationSyntax omds
+					&& omds.Modifiers.Any(m => m.IsKind(SyntaxKind.OverrideKeyword))
+					&& omds.SyntaxTree != null)
+				{
+					var methodSymbol = omds.GetSymbol(context.SemanticModel.Compilation);
+					if (methodSymbol?.Name?.Equals(CreateMauiAppMethodName) ?? false)
+					{
+						var returnTypeSymbol = context.SemanticModel.GetSymbolInfo(omds.ReturnType).Symbol;
+						var rtsName = returnTypeSymbol?.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
+						if (rtsName?.Equals(MauiAppTypeFullyQualifiedName) ?? false)
+							return true;
+					}
+				}
+			}
+
+			return false;
+		}
+
+		class MaciOSMauiAppBuilderReceiver
+		{
+			public bool HasMainMethod { get; set; } = false;
+			public bool HasAppDelegateSubclass { get; set; } = false;
+
+			public ClassDeclarationSyntax? PartialAppDelegateSubclass { get; set; } = null;
+
+			public bool PartialAppDelegateOverridesCreate { get; set; } = false;
+
+			public void OnVisitSyntaxNode(GeneratorSyntaxContext context, SyntaxNode syntaxNode)
+			{
+				var compilation = context.SemanticModel.Compilation;
+
+				// Look for a Main method
+				if (!HasMainMethod
+					&& syntaxNode is MethodDeclarationSyntax method)
+				{
+					if (method.Modifiers.Any(m => m.IsKind(SyntaxKind.StaticKeyword)))
+					{
+						var methodName = method.GetMethodName(context.SemanticModel.Compilation);
+						if (methodName?.Equals(MainMethodName) ?? false)
+						{
+							HasMainMethod = true;
+						}
+					}
+				}
+
+				// Look for an AppDelegate implementation of some sort
+				if (syntaxNode is BaseTypeSyntax baseTypeSyntax)
+				{
+					var baseTypeSymbol = context.SemanticModel.Compilation.GetSemanticModel(syntaxNode.SyntaxTree)
```

Comment:
you have the compilation at line 69
```suggestion
					var baseTypeSymbol = compilation.GetSemanticModel(syntaxNode.SyntaxTree)
```